### ISSUE TITLE
Invite Request Cooldown

### ISSUE THREAD

**Issue by Xanadeas** [Yellow-Dog-Man/Resonite-Issues#2304]
### Is your feature request related to a problem? Please describe.

At the moment there is no cooldown on invite requests making it very easy to spam someone and flood their notifications with requests.

### Describe the solution you'd like

A simple cooldown timer could be added. Something like 10-20 seconds between requests would prevent spam.

### Describe alternatives you've considered

Currently there's no alternative to prevent spam.

### Requesters

Xanadeas

**Comment by Frooxius** [comment]
There actually is cooldown right now, it's just relatively high. 

I'm not sure if this is something that really needs a technical solution on technical level. E.g. some users do like to send a few invites to get someone's attention, I don't think we want to necessarily prevent that, since it can have its uses.

Technically invites are not really different from other messages to - which could be used to spam too - but you can also legitimately send multiple messages within a 10 second interval. 

If you don't like people spamming you with it there's few things that could be done:
1) Try telling them to stop - this is simple, but can be effective. Sometimes people don't realize it's annoying someone, so communicating that can help
2) We could possibly add settings to mute invite/messages from particular person if they keep pestering you with it

**Comment by Xanadeas** [comment]
This is true, sending multiple messages or invites can grab someone's attention. But sending them en masse would be generally rude in my opinion. Thankfully I haven't run into it myself but I'm sure others have. on VRC I have multiple friends who get invite spammed sometimes in the hundreds and without a cooldown there's no real way to prevent it. I just think it'd be wise to address it before it could become a problem.

However it sounds like there's already a system in place to prevent mass spam. I just haven't encountered it personally as I don't usually spam invites or messages.

**Comment by shiftyscales** [comment]
> We could possibly add settings to mute invite/messages from particular person if they keep pestering you with it

Being able to selectively turn off different kinds of notifications on a per-user basis would be useful. Here is an example of what this looks like in Steam:
![image](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/54213390/2a8e24b8-fd1c-49fa-875d-063b28815610)

However I had figured that level of granularity would probably be better offered following the contacts UI rework #1116?

**Comment by Xanadeas** [comment]
Mass spammed no. Smaller amounts of spam yes. Having a selective notification system would be nice. But if this is something that'd go better with the UI rework I understand. I also understand if it's not a massive issue not pouring resources into it. It was just something I was worried about as I've seen it happen elsewhere and I tend to err on the side of caution.

**Comment by SlyTheFloof** [comment]
UIX spam on the invite buttons also exists still and it also works on avatar equip and causes crashes to sessons when done

hopefully that will be all fixed in a uix rework in the coming years
